File: C_by_Harry/CH-3_practice_set.c/04_problem.c
#include <stdio.h>

int main()
{
    // todo Write a program to find whether a year input by the user is leap year or not
    int year;
    printf("Enter year : \n");
    scanf("%d", &year);
    if(year<1){
        printf("year cannot be less than 1");
    }
    else if ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0) //* leap year must be divisible by both 4 and 100
    {
        printf("This is a leap year");
    }
    else
    {
        printf("Not a leap year");
    }

    return 0;
}

File: python_by_harry/CH-7 practice set/04_problem.py
'''
4. Write a program to find whether a given number is prime or not. 

'''

n = int(input("Enter a number : "))
if(n<=1):
    print("Not prime")
else:
    let_prime = True
    for x in range(2,n):
        if(n%x==0):
            let_prime = False
            break
    
    if let_prime :
        print("Prime")
    else:
        ("Not prime")
